tests/samplers/test_typical_acceptance_sampler.py

```diff
@@ -0,0 +1,401 @@
+"""Tests for rejection sampling."""
```

Comment:
These tests are great. Can we add one more which tries a different value of `posterior_threshold` and `posterior_alpha`?

Reply:
Added a test for non-default posterior_threshold and posterior_alpha